core/src/main/kotlin/fr/sncf/osrd/api/etcs/ETCSBrakingCurvesEndpoint.kt

```diff
@@ -0,0 +1,197 @@
+            val zonePath = routePath.flatMap { infra.rawInfra.getRoutePath(it) }
+            val envelopeSimPath =
+                EnvelopeTrainPath.from(infra.rawInfra, pathProps, electricalProfileMap)
+            val powerRestrictionsLegacyMap =
+                parsePowerRestrictions(request.powerRestrictions).toRangeMap()
+            val electrificationMap =
+                envelopeSimPath.getElectrificationMap(
+                    rollingStock.basePowerClass,
+                    powerRestrictionsLegacyMap,
+                    rollingStock.powerRestrictions,
+                    request.useElectricalProfiles
+                )
+            val curvesAndConditions =
+                rollingStock.mapTractiveEffortCurves(electrificationMap, request.comfort)
+            val signalingRanges = buildSignalingRanges(infra, blockPath, chunkPath)
+            val stopPositions = getStopPositions(parseRawSimulationScheduleItems(request.schedule))
+            val context =
+                EnvelopeSimContext(
+                    rollingStock,
+                    envelopeSimPath,
+                    2.0,
+                    curvesAndConditions.curves,
+                    makeETCSContext(rollingStock, infra, chunkPath, routePath, signalingRanges)
+                )
+
+            // Parse mrsp.
+            val mrsp = parseRawMrsp(request.mrsp, Offset(envelopeSimPath.length.meters))
+
+            // Compute ETCS braking curves.
+            val etcsSimulator = ETCSBrakingSimulatorImpl(context)
+            // Compute slowdown braking curves.
+            val slowdowns = etcsSimulator.computeSlowdownLocations(mrsp)
+            val slowdownBrakingCurves = etcsSimulator.computeSlowdownBrakingCurves(mrsp, slowdowns)
+            // Compute stop braking curves.
+            val stops = stopPositions.map { Offset<TravelledPath>(it.meters) }
+            val etcsStops = etcsSimulator.computeStopLocations(mrsp, stops)
+            val stopBrakingCurves = etcsSimulator.computeStopBrakingCurves(mrsp, etcsStops)
+            // Compute block transition braking curves.
+            val etcsSignalBlockPathOffsets =
+                infra.blockInfra.getSignalOffsets(infra.rawInfra, blockPath, ETCS_LEVEL2.id)
+            val travelledPathOffset =
+                Distance.ZERO -
+                    trainPathBlockOffset(infra.rawInfra, infra.blockInfra, blockPath, chunkPath)
+                        .distance
+            val etcsSignalOffsets =
+                etcsSignalBlockPathOffsets
+                    .map { Offset<TravelledPath>(it.distance - travelledPathOffset) }
+                    .filter { it.distance > Distance.ZERO && it.distance <= chunkPath.length }
+            val etcsSignalEoas = etcsSimulator.computeStopLocations(mrsp, etcsSignalOffsets)
+            val signalBrakingCurves = etcsSimulator.computeStopBrakingCurves(mrsp, etcsSignalEoas)
+
+            // Parse response.
```

Comment:
```suggestion
            // Build response.
```

Or `serialize`, or `compose`. Parsing is the opposite.

Same for the method names

Reply:
See https://github.com/OpenRailAssociation/osrd/pull/11864/commits/fcf37961c133937bc677b783f6e49151a1230439.

Reply:
That was the wrong one in the fixup :smile: It's still `parse` in the code that builds the response (l126 to 132)

Reply:
Oh right that's why you said `serialize`. Went too fast x) 
Just put `build` everywhere, as the serialization is handled by the json adapter :)
Fixed in https://github.com/OpenRailAssociation/osrd/pull/11864/commits/7f001c95dfa5eb93b0e29dca1615583d192f4be0.